03week/ticTacToe.js

```diff
@@ -1,93 +1,223 @@
-  console.log('  ---------');
-  console.log('2 ' + board[2].join(' | '));
-}
-
-function horizontalWin() {
-  // Your code here
-}
-
-function verticalWin() {
-  // Your code here
-}
-
-function diagonalWin() {
-  // Your code here
-}
-
-function checkForWin() {
-  // Your code here
+let playerTurn = "X";
+let gameover = false;
+let turn = 1;
+
+const printBoard= () => {
+  console.log("   0  1  2");
+  console.log("0 " + board[0].join(" | "));
+  console.log("  ---------");
+  console.log("1 " + board[1].join(" | "));
+  console.log("  ---------");
+  console.log("2 " + board[2].join(" | "));
 }
 
-function ticTacToe(row, column) {
-  // Your code here
+//Check for horizontal wins using arrays containing possible winning combos and checking against board state (checkForWin())
+const horizontalWin = () => {
+  const hort1 = [
+    [0, 0],
+    [0, 1], 
+    [0, 2]
+  ];
+  const hort2 = [
+    [1, 0], 
+    [1, 1], 
+    [1, 2]
+  ];
+  const hort3 = [
+    [2, 0], 
+    [2, 1], 
+    [2, 1]
+  ];
+  
+  if (checkForWin(hort1, playerTurn, board) === true) {
+    return true;
+  }
+  if (checkForWin(hort2, playerTurn, board)) {
+    return true;
+  }
+  if (checkForWin(hort3, playerTurn, board)) {
+    return true;
+  }
+    return false;
+};
+
+//Check for vertical wins using arrays containing possible winning combos and checking against board state (checkForWin())
+const verticalWin = () => { 
+  const vert1 = [
+    //0 1
+    [0, 0], //0 <-- 2nd arg in foreach
+    [1, 0], //1
+    [2, 0] //2
+  ];
+  const vert2 = [
+    [0, 1], 
+    [1, 1], 
+    [2, 1]
+  ];
+  const vert3 = [
+    [0, 2], 
+    [1, 2], 
+    [2, 2]
+  ];
+
+  if (checkForWin(vert1, playerTurn, board) == true) {
+    // console.log("win");
+    return true;
+  }
+  if (checkForWin(vert2, playerTurn, board)) {
+    return true;
+  }
+  if (checkForWin(vert3, playerTurn, board)) {
+    return true;
+  }
+    return false;
+};
+//Check for diagonal wins by using arrays containing all possible win combos and checking against board state (checkForWin())
+const diagonalWin = () => { 
+  const diag1 = [
+    //0 1
+    [0, 0], //0 <-- 2nd arg in foreach
+    [1, 1], //1
+    [2, 2] //2
+  ];
+  const diag2 = [
+    [0, 2], 
+    [1, 1], 
+    [2, 0]
+  ];
+  if (
+    checkForWin(diag1, playerTurn, board) ||
+    checkForWin(diag2, playerTurn, board)
+  ) {
+    return true;
+  }
+    return false;
+};
+
+//Checking the board state based on an array of conditions passed in
+const checkForWin = (wincondition, turn, board) => {
+  let winner =true;
+  wincondition.forEach(conditions => {
+    // Board      row                col
+    if (board[conditions[0]][conditions[1]] !== playerTurn) {
+      winner = false;
+    }
+  });
+    return winner;
+};
+
+const switchPlayer = () => {
+  if (playerTurn === "X") {
+    playerTurn = "O";
+  } else {
+    playerTurn = "X";
+  }
+};
+
+//Check all winning combinations and return true if any of them return true
+const isWinner = () => {
```

Comment:
this should be the checkForWin, you are over complicating with your individual win functions.